### PR TITLE
Duplo 15256 Ecache instance snapshot support

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview
Added support for attribute `snapshot_name`, `snapshot_arns`, `snapshot_retention_limit` for `duplocloud_ecache_instance`

## Summary of changes

This PR does the following:

- Added snapshot attribute snapshot_name, snapshot_arns (conflicts with snapshot_name) and snapshot_retention_limit 
- Updated docs

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [✓ ] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-15256 TF: duplocloud_ecache_instance: Add support to create redis with snapshot and `Snapshot Retention Limit` field](https://app.clickup.com/t/8655600/DUPLO-15256)

File: duplocloud/resource_duplo_ecache_instance.go [hunk 1]
```diff
@@ -157,6 +157,27 @@ func ecacheInstanceSchema() map[string]*schema.Schema {
...
 		},
+		"snapshot_arns": {
+			Description:   "Specify the ARN of a Redis RDB snapshot file stored in Amazon S3. User should have the access to export snapshot to s3 bucket. One can find steps to provide access to export snapshot to s3 on following link https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/backups-exporting.html",
+			Type:          schema.TypeList,
+			Optional:      true,
+			Computed:      true,
+			ConflictsWith: []string{"snapshot_name"},
+			Elem:          &schema.Schema{Type: schema.TypeString},
+		},
+		"snapshot_name": {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_ecache_instance.go:168]: We should add ConflictsWith: []string{"snapshot_arns"}, as well on this attribute.

File: duplocloud/resource_duplo_ecache_instance.go [hunk 2]
```diff
@@ -355,6 +380,9 @@ func flattenEcacheInstance(duplo *duplosdk.DuploEcacheInstance, d *schema.Resour
 	d.Set("parameter_group_name", duplo.ParameterGroupName)
 	d.Set("enable_cluster_mode", duplo.EnableClusterMode)
 	d.Set("number_of_shards", duplo.NumberOfShards)
+	d.Set("snapshot_name", duplo.SnapshotName)
+	d.Set("snapshot_arn", duplo.SnapshotArn)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_ecache_instance.go:384]: Use `snapshot_arns` instead of `snapshot_arn`

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_ecache_instance.go]: made changes

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**